internal/controller/isbservicerollout_controller.go

```diff
@@ -234,11 +234,17 @@ func processISBServiceStatus(ctx context.Context, isbsvc *kubernetes.GenericObje
 	isbSvcPhase := numaflowv1.ISBSvcPhase(isbsvcStatus.Phase)
 	switch isbSvcPhase {
 	case numaflowv1.ISBSvcPhaseFailed:
-		rollout.Status.MarkChildResourcesUnhealthy("ISBSvcFailed", "ISBService Failed")
+		rollout.Status.MarkChildResourcesUnhealthy("ISBSvcFailed", "ISBService Failed", rollout.Generation)
+	case numaflowv1.ISBSvcPhasePending:
+		rollout.Status.MarkChildResourcesUnhealthy("ISBSvcPending", "ISBService Pending", rollout.Generation)
 	case numaflowv1.ISBSvcPhaseUnknown:
-		// this will have been set to Unknown in the call to InitConditions()
+		rollout.Status.MarkChildResourcesHealthUnknown("ISBSvcUnknown", "ISBService Phase Unknown", rollout.Generation)
 	default:
-		rollout.Status.MarkChildResourcesHealthy()
+		if isbsvc.Generation == isbsvcStatus.ObservedGeneration {
```

Comment:
why not checking ObservedGeneration is not empty first? Since it can be absent for existing numaflow versions.

Reply:
I'm a little confused here - I thought we didn't need to worry about backward compatibility for older versions of Numaflow after all?

Reply:
sorry for the confusion, no worry about backward compatibility for older version to me means: 1) we don't need to ensure we can correctly detect if the changes has been applied for existing numaflow versions that does not have `observedGeneration` in status yet 2) we should not introduce any bugs if someone is running existing numaflow versions. If we have this current checking, with existing numaflow version it will never be healthy, right?

Reply:
In that case, I guess the code won't truly be able to detect the difference between "this version of numaflow doesn't support observedGeneration" and "Numaflow Controller slow/down" for a newly created resource

Reply:
@xdevxy and I discussed and we think that [this](https://github.com/numaproj/numaplane/pull/92/commits/2728f241b2c600322e76d4924fae524e2fb12573) should address that concern. The comment explains that we assume that Numaflow will set default ObservedGeneration to -1. So, when the generation goes from 0 to 1, the ObservedGeneration will be set to 1 skipping the zero value.

Reply:
which code you are referring to? this one? https://github.com/numaproj/numaplane/pull/92/files#diff-9a10dece0454f1f44e2e6076e2b3ada6a08a66de674f546276972944ec61be9aR265-R268. Only "version of numaflow doesn't support observedGeneration" can ever be 0 in this case, right?

Reply:
> @xdevxy and I discussed and we think that [this](https://github.com/numaproj/numaplane/pull/92/commits/2728f241b2c600322e76d4924fae524e2fb12573) should address that concern. The comment explains that we assume that Numaflow will set default ObservedGeneration to -1. So, when the generation goes from 0 to 1, the ObservedGeneration will be set to 1 skipping the zero value.

That is a creative idea. So, this would be the default version set in the CRD itself I guess?

Reply:
yes